bot/controllers/welcome.js

```diff
@@ -0,0 +1,34 @@
+const BaseController = require('../baseController.js');
+const Command = require('../baseCommand.js');
+const util = require('apex-util');
+
+class WelcomeController extends BaseController {
+  constructor(message) {
+    // Call BaseController constructor
+    super(message);
+
+    // Aliasing 'this' as controller to allow for binding in actions
+    const controller = this;
+
+    // Array of all commands, see baseCommand.js for prototype
+    this.commands = [
+      new Command(
+        '!welcome',
+        '!welcome',
+        'Send Welcome Message',
+        'Sends A Direct Welcome Message.',
+        this.welcomeAction.bind(controller),
+        'dm',
+      ),
+    ];
+  }
+
+  // Sends the welcome message to the user
+  welcomeAction() {
+    const { message } = this;
+    util.log('Sending Welcome Message to User', 0);
```

Comment:
Should this be a level 0 debug event?

Reply:
Changed this around and added it to a level 3 log, as I believe it has to do with an extended controller action, rather than an action close to the top level.